lib/rack/params/context.rb

```diff
@@ -125,34 +111,68 @@ class HashContext < Context
       # do type coercion and validation for a parameter with the given key.
       # adds the coerced value to the result hash, or push an error.
       #
-      # @see #_coerce #_coerce defines valid types for coercion.
-      # @param key the key in the hash of the parameter to validate
-      # @param type the type to coerce the value into
-      # @param options [Hash]
-      # @yield
-      #   if type is {Hash} or {Array}, passing a block will recursively
-      #   validate the coerced value, assuming the block is a new validation
-      #   context.
+      # @param [String] key the key in the parameter hash
+      # @param [Hash] options
+      # @option options [Boolean] :required boolean will fail if key is missing from params
+      # @option options [Any]     :default the value to return if the param is missing
+      # @option options [Boolean] :allow_nil   is nil considered present?, defaults to false, considered missing.
+      # @option options [Boolean] :allow_blank is blank (0-length string, or nil) considered present?, defaults to false, considered missing.
       # @return the coerced value
-      def param(key, type, options = {}, &block)
+      #
+      # @overload param(key, options = {}, &block)
+      #   coerces the value through the block given, marking invalid on falsey and raised errors.
+      #   (use the `allow_nil` and `allow_falsey` options to change the validity behavoir)
```

Comment:
actually write support for `allow_nil` and `allow_falsey`, as well as specs that test for invalid parsing in the block (never did that).